CFW_loader/source/main.c

```diff
@@ -7,226 +7,191 @@
 #include "platform.h"
 #include "draw.h"
 
-//VARIABLES
-char* systemVersion;
-char type;
-void ioDelay(u32);
-bool dump_arm9;
-
-void ClearTop() {
-	ClearScreen(TOP_SCREEN0, RGB(0, 0, 0));
-	ClearScreen(TOP_SCREEN1, RGB(0, 0, 0));
-	ClearScreen(BOT_SCREEN0, RGB(0, 0, 0));
-	ClearScreen(BOT_SCREEN1, RGB(0, 0, 0));
-	current_y = 0;
-}
-
-void wait_key() {
-	Debug("");
-	Debug("Press key to continue");
-	InputWait();
+//This contains the System Firmware Version String.
+char* cfw_FWString;
+//This is the System Firmware Version in system.txt
+char cfw_FWValue;
+//if true, dump the ARM9 ram
+bool cfw_arm9dump;
+
+//[Unused]
+//void ioDelay(u32);
+
+// @breif  [Unused]Simply wait for a key, but not use its value.
+// @note   Work like a pause.
+void CFW_WaitKey(void) {
+	DrawDebug(1,"");
+	DrawDebug(1,"Press key to continue");
+	HidWaitForInput();
 }
 
-void getSystemVersion()
-{
+// @breif  This reads the system version from the configuration file.
+// @note   So please make sure you've have the file in your SD.
+//         Also, if we use binary data to store the system version..
+void CFW_getSystemVersion(void) {
 	char settings[16];
-	if (FileOpen("/3ds/PastaCFW/system.txt")){
-		FileRead(settings, 16, 0);
-		FileClose();
+	if (FSFileOpen("/3ds/PastaCFW/system.txt")){
+		FSFileRead(settings, 16, 0);
+		FSFileClose();
 	}
-
-	switch (settings[0])
-	{
-	case '1': // 4.x
-		type = '1';
-		systemVersion = "Old 3DS V. 4.1 - 4.5";
-		break;
-	case '2': // 5.0
-		type = '2';
-		systemVersion = "Old 3DS V. 5.0";
-		break;
-	case '3': // 5.1
-		type = '3';
-		systemVersion = "Old 3DS V. 5.1";
-		break;
-	case '4': // 6.0
-		type = '4';
-		systemVersion = "Old 3DS V. 6.0";
-		break;
-	case '5': // 6.1 - 6.3
-		type = '5';
-		systemVersion = "Old 3DS V. 6.1 - 6.3";
-		break;
-	case '6': // 7.0-7.1
-		type = '6';
-		systemVersion = "Old 3DS V. 7.0 - 7.1";
-		break;
-	case '7': // 7.2
-		type = '7';
-		systemVersion = "Old 3DS V. 7.2";
-		break;
-	case '8': // 8.x
-		type = '8';
-		systemVersion = "Old 3DS V. 8.0 - 8.1";
-		break;
-	case '9': // 9.x
-		type = '9';
-		systemVersion = "Old 3DS V. 9.0 - 9.2";
-		break;
-	case 'a': // 8.x
-		type = 'a';
-		systemVersion = "New 3DS V. 8.1";
-		break;
-	case 'b': // 9.x
-		type = 'b';
-		systemVersion = "New 3DS V. 9.0 - 9.2";
-		break;
+    cfw_FWValue = settings[0];
+	switch (settings[0]) {
```

Comment:
If we adjust how the `cfw_FWValue` store its value, this convertion would be done much easier. *Example* `'1'` to binary `0x01` or `'1'` to `'a'`.
Also a marker telling which version the configuration file may be introduced.